pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -188,10 +197,30 @@ func (pc *PodController) enqueuePod(obj interface{}) {
 			return
 		}
 	}
-	podKey := podKeyGet(pod)
+	podKey := podKeyGet(pod.Namespace, pod.Name)
 	pc.queue.Add(podKey)
 }
 
+// processCNIUpdate will be called when CNIServer publishes a Pod update event.
+func (pc *PodController) processCNIUpdate(e interface{}) {
+	event := e.(types.PodUpdate)
+
+	if event.IsAdd {
+		// Go cache the CNI server info at CNIConfigInfo cache, for podWatch usage
+		cniInfo := &cnipodcache.CNIConfigInfo{PodName: event.PodName, PodNamespace: event.PodNamespace,
+			ContainerID: event.ContainerID, ContainerNetNS: event.NetNS, PodCNIDeleted: false}
+		pc.podCache.AddCNIConfigInfo(cniInfo)
+	} else {
+		containerInfo := pc.podCache.GetCNIConfigInfoByContainerID(event.PodName, event.PodNamespace, event.ContainerID)
+		if containerInfo != nil {
+			// Update PodCNIDeleted = true.
+			// This is to let Podwatch controller know that the CNI server cleaned up this Pod's primary network configuration.
+			pc.podCache.SetPodCNIDeleted(containerInfo)
+		}
+	}
+	pc.queue.Add(podKeyGet(event.PodNamespace, event.PodName))
```

Comment:
I guess this is a key code difference in this patch: with CNI events being handled asynchronously, we need to ensure we enqueue the Pods